tests/src/Kernel/Entity/FieldAccessTest.php

```diff
@@ -0,0 +1,190 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\Tests\og\Kernel\Entity\FieldAccessTest.
+ */
+
+namespace Drupal\Tests\og\Kernel\Entity;
+
+use Drupal\Core\Session\AnonymousUserSession;
+use Drupal\KernelTests\KernelTestBase;
+use Drupal\og\Og;
+use Drupal\Component\Utility\Unicode;
+use Drupal\node\Entity\NodeType;
+use Drupal\user\Entity\User;
+use Drupal\user\Entity\Role;
+
+/**
+ * Tests OG specific field access.
+ *
+ * @group og
+ */
+class FieldAccessTest extends KernelTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static $modules = ['system', 'user', 'field', 'entity_reference', 'node', 'og'];
+
+  /**
+   * @var string
+   *
+   * The machine name of the group node type.
+   */
+  protected $groupBundle;
+
+  /**
+   * @var \Drupal\user\Entity\User
+   */
+  protected $adminUser;
+
+  /**
+   * @var \Drupal\user\Entity\User
+   */
+  protected $authenticatedUser;
+
+  /**
+   * @var \Drupal\Core\Entity\EntityAccessControlHandlerInterface
+   */
+  protected $userAccessControlHandler;
+
+  /**
+   * The audience field definition.
+   *
+   * @var \Drupal\field\Entity\FieldConfig
+   */
+  protected $fieldDefinition;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    // Add membership and config schema.
+    $this->installConfig(['og']);
+    $this->installEntitySchema('og_membership');
+    $this->installEntitySchema('user');
+    $this->installEntitySchema('node');
+    $this->installSchema('system', 'sequences');
+
+    // Setting up variables.
+    $this->groupBundle = Unicode::strtolower($this->randomMachineName());
+
+    // Create a group.
+    NodeType::create([
+      'type' => $this->groupBundle,
+      'name' => $this->randomString(),
+    ])->save();
+
+    // Define the group content as group.
+    Og::groupManager()->addGroup('node', $this->groupBundle);
```

Comment:
I'd prefer this access test to use node only when there's node specific requirements. We should test on a "test entity" to confirm we are not building node-centric stuff.